examples/src/main/java/com/cloudera/nav/sdk/examples/lineage3/LineageExport.java

```diff
@@ -0,0 +1,219 @@
+  private final static String LINEAGE_EXPORT_API = "lineage3/export";
+  private final static String LINEAGE_EXPORT_API_QUERY_PARAMS = "?direction=%s&length=%s&lineageOptions=%s";
+  private final static String DEFAULT_FILENAME = "lineage.json";
+  private final static String IDENTITY = "identity";
+  private final String charset = StandardCharsets.UTF_8.name();
+
+  /**
+   * @param args 1. config file path
+   *             2. query to fetch entities for which lineage
+   *             would be downloaded.
+   *             3. direction
+   *             4. length
+   *             5. lineage options
+   * @throws Exception
+   */
+  public static void main(String[] args) throws Exception {
+    Preconditions.checkArgument(args.length == 5);
+    Preconditions.checkArgument("downstream".equals(args[2].toLowerCase())
+        || "upstream".equals(args[2].toLowerCase()));
+    Preconditions.checkArgument(Integer.parseInt(args[3]) > 0);
+    Preconditions.checkArgument(Integer.parseInt(args[4]) >= 0);
+
+    (new LineageExport(args[0], args[1], args[2].toUpperCase(), args[3], args[4])).run();
+  }
+
+  public LineageExport(String configPath, String query, String direction,
+                       String length, String lineageOptions) {
+    this.configPath = configPath;
+    this.query = query;
+    this.direction = direction;
+    this.length = length;
+    this.lineageOptions = lineageOptions;
+  }
+
+  public void run() throws IOException {
+    NavApiCient client = NavigatorPlugin.fromConfigFile(configPath)
+        .getClient();
+
+    String cursorMark = null;
+    List<Map<String, Object>> results = null;
+    ResultsBatch<Map<String, Object>> rb;
+    FileWriter writer = new FileWriter(DEFAULT_FILENAME);
+
+    do {
+      MetadataQuery mq = new MetadataQuery(query, 100, cursorMark);
+      rb = client.getEntityBatch(mq);
+
+      results = rb.getResults();
+      cursorMark = rb.getCursorMark();
+
+      // extract out all the entityIds for which to download
+      // lineage for
+      Set<String> entityIds = Sets.newHashSet();
+      for (Map<String, Object> entities : results) {
+        entityIds.add((String) entities.get(IDENTITY));
+      }
+
+      if (entityIds.size() > 0) {
+        fetchAndWriteLineage(entityIds, writer, client.getConfig());
+      }
+    } while (results.size() != 0);
+
+    writer.flush();
+    writer.close();
+  }
+
+  @SuppressWarnings("unchecked")
+  private void fetchAndWriteLineage(Set<String> entityIds, FileWriter fw,
+                                    ClientConfig config) throws IOException {
+    HttpURLConnection connection = createHttpConnection(config, entityIds);
+
+    InputStream stream = connection.getInputStream();
+
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    mapper.configure(JsonParser.Feature.ALLOW_COMMENTS, true);
+    ArrayList entities = (ArrayList) mapper.readValue(stream, Object.class);
```

Comment:
You can use new new TypeReference<Map<String, Object>>() {} instead of Object.class here. That way you can remove the @SuppressWarnings and the raw ArrayList/Iterator usage